### DISCUSSION TITLE
iOS libpag4.3.33 崩溃

### DISCUSSION THREAD

**Discussion by mwxia** [Tencent/libpag#2313] [Q&A] [answered]
libpag 版本信息: 4.3.33 （pod 'libpag', :git => 'https://github.com/libpag/libpag-ios.git', :tag => '4.3.33'）
系统版本分布情况 : 目前仅收集到崩溃发生在iOS 17系统上
是否存在复现路径: 线上崩溃，未复现

代码如下：
![image](https://github.com/Tencent/libpag/assets/9984670/fbc320c8-78f8-4e80-8f90-d874188347d3)

排查日志发现，每次崩溃发生前，都会报如下错误，不知道和这个有没有关系：
file open failed! filePath:/var/mobile/Containers/Data/Application/33A65B90-B4FE-4CD9-8744-9EED3E32B983/Library/Caches/libpag/files/1.bin 

堆栈详情如下:
1. 
Crashed: com.google.firebase.crashlytics.ios.exception
0  FirebaseCrashlytics            0x1d068 FIRCLSProcessRecordAllThreads + 393 (FIRCLSProcess.c:393)
1  FirebaseCrashlytics            0x1d448 FIRCLSProcessRecordAllThreads + 424 (FIRCLSProcess.c:424)
2  FirebaseCrashlytics            0x146a8 FIRCLSHandler + 34 (FIRCLSHandler.m:34)
3  FirebaseCrashlytics            0xf3b4 __FIRCLSExceptionRecord_block_invoke + 232 (FIRCLSException.mm:232)
4  libdispatch.dylib              0x3dd4 _dispatch_client_callout + 20
5  libdispatch.dylib              0x132c4 _dispatch_lane_barrier_sync_invoke_and_complete + 56
6  FirebaseCrashlytics            0xe3c0 FIRCLSExceptionRecord + 234 (FIRCLSException.mm:234)
7  FirebaseCrashlytics            0xeed4 FIRCLSExceptionRecordNSException + 126 (FIRCLSException.mm:126)
8  FirebaseCrashlytics            0xe018 FIRCLSTerminateHandler() + 398 (FIRCLSException.mm:398)
9  libc++abi.dylib                0x14068 std::__terminate(void (*)()) + 16
10 libc++abi.dylib                0x1735c __cxa_get_exception_ptr + 86
11 libc++abi.dylib                0x172a0 __cxxabiv1::exception_cleanup_func(_Unwind_Reason_Code, _Unwind_Exception*) + 306
12 libobjc.A.dylib                0x16180 objc_exception_throw + 420
13 CoreFoundation                 0x18d480 +[NSObject(NSObject) _copyDescription] + 342
14 CoreFoundation                 0x20fb4 ___forwarding___ + 1572
15 CoreFoundation                 0x208d0 _CF_forwarding_prep_0 + 96
16 libpag                         0xb6c2c pag::PAGDecoder::setCacheKeyGeneratorFun(std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> (pag::PAGDecoder*, std::__1::shared_ptr<pag::PAGComposition>)>) + 2844
17 libpag                         0xb9594 pag::PAGFile::fileFrameToScaledFrame(long long, pag::TimeRange const&) const + 2036
18 libpag                         0xbada4 pag::PAGFile::fileFrameToScaledFrame(long long, pag::TimeRange const&) const + 8196
19 libpag                         0x10cd88 pag::StrokeStyle::verify() const + 9604
20 libpag                         0x10d11c pag::StrokeStyle::verify() const + 10520
21 libpag                         0x10db38 pag::StrokeStyle::verify() const + 13108
22 libsystem_pthread.dylib        0x606c _pthread_start + 136
23 libsystem_pthread.dylib        0x10d8 thread_start + 8
    
2.
Crashed: Thread
0  libobjc.A.dylib                0x3c20 objc_msgSend + 32
1  libpag                         0xb6c2c pag::PAGDecoder::setCacheKeyGeneratorFun(std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> (pag::PAGDecoder*, std::__1::shared_ptr<pag::PAGComposition>)>) + 2844
2  libpag                         0xb9594 pag::PAGFile::fileFrameToScaledFrame(long long, pag::TimeRange const&) const + 2036
3  libpag                         0xbada4 pag::PAGFile::fileFrameToScaledFrame(long long, pag::TimeRange const&) const + 8196
4  libpag                         0x10cd88 pag::StrokeStyle::verify() const + 9604
5  libpag                         0x10d11c pag::StrokeStyle::verify() const + 10520
6  libpag                         0x10db38 pag::StrokeStyle::verify() const + 13108
7  libsystem_pthread.dylib        0x2a90 _pthread_start + 136
8  libsystem_pthread.dylib        0x1fcc thread_start + 8

**Comment by guihuaYi** [comment]
我们也遇到这个问题了，请问解决了吗

**Comment by kevingpqi123** [comment, marked as answer]
建议升级至最新版本 4.3.57
4.3.33 是 4.3 的第一个版本，一个版本系列我们会不断完善修复问题，不会新增功能，出现问题第一时间升级版本

**Comment by zhaoqt521** [comment]
iPhone13Pro  系统版本 18.7.0  接入libpag版本是4.4.35 
目前也偶现出现了如下错误  暂时没有收到更新的堆栈信息
file open failed! filePath:/var/mobile/Containers/Data/Application/8E535EDE-D24E-4D75-9E3B-D4FAD36E0548/Library/Caches/libpag/files/1335.bin

**Reply by kevingpqi123** [reply]
这个只是一个报错，如果读取缓存失败，会重新构建缓存，不会引起渲染异常
